lib/src/logic/validators.dart

```diff
@@ -264,7 +266,11 @@ class ShouldInBetweenDatesValidator<KeyType>
 
   @override
   bool isValid(DateTime date) {
-    return (date.compareTo(min) > 0 && date.compareTo(max) < 0);
+    if(isInclusive){
+      return (date.isBetween(min, max));
```

Comment:
Having another look at the `compareTo` I think the logic here could be much simple, we don't need to implement this new `isBetween` method. We could just use `return (date.compareTo(min) >= 0 && date.compareTo(max) <= 0);`.

And to fix the broken exclusive case we could use `return (date.compareTo(min) > 0 && date.compareTo(max) < 0);`